src/OrchardCore/OrchardCore.Environment.Shell/ShellHost.cs

```diff
@@ -88,6 +89,33 @@ public ShellContext GetOrCreateShellContext(ShellSettings settings)
             return shell;
         }
 
+        public IServiceScope EnterServiceScope(ShellSettings settings)
+        {
+            return EnterServiceScope(settings, out var context);
+        }
+
+        public IServiceScope EnterServiceScope(ShellSettings settings, out ShellContext context)
+        {
+            context = GetOrCreateShellContext(settings);
```

Comment:
I suppose the `ShellContext` might be released immediately after this method is called (and before `TryEnterServiceScope` has a chance to be invoked).

Should `ShellHost` use good old `locks` for all its operations?